Ginger/Ginger/BusinessFlowPages_New/AddActionMenu/MainAddActionsNavigationPage.xaml.cs

```diff
@@ -95,5 +123,76 @@ private void LoadActionFrame(Page navigationPage, string titleText = "", eImageT
                 xSelectedItemTitlePnl.Visibility = Visibility.Collapsed;
             }
         }
+
+        //public void StartAgent()
+        //{
+        //    IsAgentStarted = false;
+        //    xRecordItemBtn.IsEnabled = false;
+        //    if (mContext == null)
+        //        return;
+
+        //    @AppAgentAct:
+        //    Activity mActParentActivity = mContext.BusinessFlow.CurrentActivity;
+        //    if (string.IsNullOrEmpty(mActParentActivity.TargetApplication))
+        //    {
+        //        if (mContext.BusinessFlow.TargetApplications.Count() == 1)
+        //        {
+        //            mActParentActivity.TargetApplication = ((ApplicationAgent)mContext.Runner.ApplicationAgents[0]).AppName; 
+        //        }
+        //    }            
+        //    ApplicationAgent appAgent = (ApplicationAgent)mContext.Runner.ApplicationAgents.Where(x => x.AppName == mActParentActivity.TargetApplication).FirstOrDefault();
+
+        //    if (appAgent == null)
+        //    {
+        //        mContext.Runner.SolutionAgents = WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<Agent>();
+        //        mContext.Runner.UpdateApplicationAgents();
+        //        goto AppAgentAct;
+        //    }
+
+        //    PlatformInfoBase platform = PlatformInfoBase.GetPlatformImpl(appAgent.Agent.Platform);
+
+        //    ObservableList<DataSourceBase> dSList = WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<DataSourceBase>();
+        //    if (appAgent != null)
+        //    {
+        //        if (appAgent.Agent.Driver == null)
+        //        {
+        //            appAgent.Agent.DSList = dSList;
+        //            appAgent.Agent.StartDriver();
+        //            xRecordItemBtn.IsEnabled = true;
+        //            IsAgentStarted = true;
+        //        }
+        //        else if (!appAgent.Agent.Driver.IsRunning())
+        //        {
+        //            if (Reporter.ToUser(eUserMsgKey.PleaseStartAgent, eUserMsgOption.OKCancel, eUserMsgSelection.OK) == eUserMsgSelection.OK)
+        //            {
+        //                appAgent.Agent.StartDriver();
+        //                xRecordItemBtn.IsEnabled = true;
+        //                IsAgentStarted = true;
+        //            }
+        //            else
+        //            {
+        //                IsAgentStarted = false;
+        //                return;
+        //            }
+        //        }
+        //        DriverBase driver = appAgent.Agent.Driver;
+        //        if (driver is IWindowExplorer)
+        //        {
+        //            mWindowExplorerDriver = (IWindowExplorer)appAgent.Agent.Driver;
+        //        }
+        //    }
+        //    else
+        //    {
+        //        IsAgentStarted = false;
+        //    }         
+        //}
+
+        public void StopRecording()
```

Comment:
StopRecording() should not be here, please remove.

Reply:
it is required when the page is switched, if the recording is on then to stop the recording.